### ISSUE TITLE
Cannot solve the problem of password in Chinese

### ISSUE THREAD

**Issue by 1657381970** [srikanth-lingala/zip4j#282]
When encrypting, set the Chinese password, and it will always prompt the password error when decrypting. I tried to convert the password into other codes, but also always prompt the password error，The version used is zip4j2.6.1。The unzip password is 你好  。Below is my code and zip file
[测试.zip](https://github.com/srikanth-lingala/zip4j/files/5901557/default.zip)
。

@Test
    public void testZip4j() throws Exception {
        String zipFile ="F:\\a\\a\\测试.zip";
        String destDir ="F:\\a\\a\\";
        String password = "你好";
        ZipFile zip = new ZipFile(zipFile, password.toCharArray());
        zip.extractAll(destDir);
    }

**Comment by srikanth-lingala** [comment]
I tried to reproduce this issue, but with the code below, I was able to extract the zip file generated with zip4j. 

```
@Test
public void testIssueAdd() throws IOException {
  File fileToAdd = new File(BASE_DIR + "123.pdf");
  ZipParameters zipParameters = new ZipParameters();
  zipParameters.setEncryptFiles(true);
  zipParameters.setEncryptionMethod(EncryptionMethod.ZIP_STANDARD);
  new ZipFile(BASE_DIR + "test.zip", "你好".toCharArray()).addFile(fileToAdd, zipParameters);

  ZipFile zipFile = new ZipFile(BASE_DIR + "test.zip", "你好".toCharArray());
  zipFile.extractAll(BASE_DIR);
}
```

I tried with `ZIP_STANDARD` as well as `AES` and both worked fine. I was not able to extract the zip file you attached, but I cannot say why that is not working. But Chinese passwords should work with zip4j. I think there are also integration tests to verify them. The only difference is that I used the latest code (v2.6.4).

**Comment by 1657381970** [comment]
I know why you can't unzip my zip package. My zip package is compressed with WinRAR tool.Strangely enough, it works when the password is set to numbers or letters, but it doesn't work when the password is in Chinese.I used Zip4j to add a ZIP file, and the password was in Chinese. I tried to unzip it with WinRAR tool, but it showed me that the password was wrong. I think the root reason should be that your algorithm is different.Is there any way to decompress WinRAR compressed files?It would help me a lot.

**Comment by srikanth-lingala** [comment]
I have identified the issue. I will work on a fix and include this in the next release.

**Comment by 1657381970** [comment]
When I use WinRAR to encrypt the file, and the password is in Chinese, in the case of knowing the password, can your zip4j decrypt?

**Comment by srikanth-lingala** [comment]
Yes, I think it should work. I do not have WiinRAR, but I can reproduce the behaviour with Keka on Mac. After converting your password to utf-8 it works fine with Keka. I think WinRAR should have the same behaviour.

**Comment by srikanth-lingala** [comment]
I have made a snapshot release with the fix. Please test with version `2.6.5-SNAPSHOT` and confirm if this works. I have tested it on mac with Chinese password (from your code) and it works fine now with other tools (Keka, The Unarchiver, etc). I will include this fix in the next release. If you have trouble getting the snapshot version, please check [this][1] answer.

[1]: https://stackoverflow.com/questions/7715321/how-to-download-snapshot-version-from-maven-snapshot-repository

**Comment by 1657381970** [comment]
I have installed the snapshot into my local Maven warehouse for testing, but the previous problem still exists. Do you have a Windows environment to test it?Can you unzip my zip after you fix the program?

**Comment by srikanth-lingala** [comment]
I tried it on a windows machine and it works fine with WinRaR now after the fix. As the bug is during the creation of the zip file, please note that a zip file already created with a previous version of zip4j cannot be extracted anymore. The zip file has to be created with the new snapshot version of zip4j. So, the zip file attached to this issue cannot be extracted anymore as this was created before the fix.

As mentioned, I created the zip file with the snapshot version of zip4j and was successfully able to extract it with WinRaR.

**Comment by 1657381970** [comment]
Is the first use of WinRAR Chinese password encryption, and then use Zip4J decompression can not.

**Comment by srikanth-lingala** [comment]
After some playing around and some researching, I think that this is a bug in WinRaR. I created a zip file with `你好` as password in WinRaR, then I tried to extract it with the same password from zip4j, and it did not work and I got a `Wrong Password` exception. But after reading about a [bug in WinRaR][1], I replaced the password with `??` (one question mark for each character), it worked fine. I think WinRaR does not handle non-ascii passwords well when creating the zip file.

[1]: http://anrieff.net/abs/viewpost.php?lang=en&id=40
